File: Lab_1/banking/BankAccount.java
package banking;
/**
 * This class represents a bank account. 
 * @author Tanner Yilmaz
 * @version 1.0
 */
public class BankAccount{
	
	protected String name;
	protected double balance;
	
	public BankAccount() {
		name = "default_name";
		balance = 0.0;
	}
	
	public BankAccount(String name, double initBalance) {
		this.name = name;
		this.balance = initBalance;
	}
	
	/**
	 * Transfers money to accounts. 
	 * @param depositAmt Amount being deposited
	 */
	public void deposit(double depositAmt) {
		setBalance(balance += depositAmt);
	}
	
	/**
	 * Withdraws money from accounts. 
	 * @param withdrawAmt Amount being withdrawn
	 */
	public void withdraw(double withdrawAmt) {
		setBalance(balance -= withdrawAmt);
	}
	
	/** Transfers money out of this account.
	 * @param receivingAccount is the account receiving the transfer.
	 * @param transferAmt is the amount being transferred.
	 */
	public void transfer(BankAccount receivingAccount, double transferAmt) {
		withdraw(transferAmt);
		receivingAccount.deposit(transferAmt);
	}
	
	public String getName() {
		return name;
	}
	
	public double getBalance() {
		return balance;
	}
	
	public void setBalance(double newBalance) {
		balance = newBalance;
	}
	
	@Override
	public String toString() {
		return "Account holder name: " + name + "\nAccount Balance: " + balance;
	}
}

File: Lab_5/Source_Code/BankAccount.cpp
#include "BankAccount.h"
#include <string>
#include <sstream>
#include <iostream>
using namespace std;

string BankAccount::getName(){
  return name;
}

double BankAccount::getBalance(){
  return balance;
}

void BankAccount::deposit(double amount){
  balance += amount;
}

void BankAccount::withdraw(double amount){
  balance -= amount;
}

string BankAccount::toString(){

  ostringstream strs; //These three lines of code convert the balance variable to a string
  strs << balance;
  string str_balance = strs.str();

  return "Name: " + name + ", Balance: " + str_balance;
}